server/contexts/token/token.go

```diff
@@ -20,14 +20,17 @@ type Token string
 func FromHTTPRequest(r *http.Request) Token {
 	headers := r.Header.Get("Authorization")
 	headerParts := strings.Split(headers, " ")
-	if len(headerParts) != 2 || strings.ToUpper(headerParts[0]) != "BEARER" {
-		if err := r.ParseForm(); err != nil {
-			return ""
+	if len(headerParts) > 0 && strings.ToUpper(headerParts[0]) == "BEARER" {
+		if len(headerParts) == 2 {
+			return Token(headerParts[1])
 		}
-
-		return Token(r.FormValue("token"))
+		// This indicates "no token". We don't want to read the request-body here.
+		return ""
+	}
+	if err := r.ParseForm(); err != nil {
+		return ""
```

Comment:
Changes look good, I wonder if it's ok for us to ignore error here? (should we at least log it?).